asdf/_node_info.py

```diff
@@ -282,7 +282,13 @@ def from_root_node(cls, key, root_identifier, root_node, schema=None, refresh_ex
 
                     if cls.traversable(node):
                         t_node = node.__asdf_traverse__()
-                        info.set_schema_from_node(node, extension_manager)
+                        if hasattr(node, "_tag") and isinstance(node._tag, str):
+                            try:
+                                info.set_schema_from_node(node, extension_manager)
+                            except KeyError:
+                                # if _tag is not a valid tag, no schema will be found
+                                # and a KeyError will be raised
+                                pass
```

Comment:
should this be raised instead, if `_tag` in invalid, to inform the user?
```suggestion
                                raise
```

Reply:
Thanks for taking a look and asking the question.

`pass` here allows `info` to succeed if the tree contains an object with a `_tag` attribute that is not a valid ASDF tag (for example if the object uses `_tag` for non-ASDF reasons). With the following example:
```python
import asdf
class Foo:
    def __asdf_traverse__(self):
        return ["bar"]
af = asdf.AsdfFile()
af['foo'] = Foo()
af.info()
```
Running the example on main fails (see #1738)
```
AttributeError: 'Foo' object has no attribute '_tag'
```
I don't believe there should be an expectation that an object that implements `__asdf_traverse__` also have a valid asdf tag in `_tag` since the [docs](https://asdf.readthedocs.io/en/latest/asdf/extending/extensions.html#making-converted-object-s-contents-visible-to-info-and-search) don't mention the need for `_tag`. With this PR the above example succeeds producing an info output:
```
root (AsdfObject)
└─foo (Foo)
  └─0 (str): bar
```
However, if we modify the example and add a `_tag` attribute to `Foo` (let's assume `Foo` uses `_tag` for some non-ASDF use) the situation depends on the `pass` vs `raise`.
```python
import asdf
class Foo:
    def __asdf_traverse__(self):
        return ["bar"]
    _tag = "non_asdf_tag"
af = asdf.AsdfFile()
af['foo'] = Foo()
af.info()
```
With the `pass` use, the `info` output succeeds and is the same as above. However, switching to a `raise` causes the `info` to fail with the raised exception. That would mean that any object implementing `__asdf_traverse__` must also have a valid ASDF tag in `_tag` if it uses the `_tag` attribute. If we want to follow that expectation the docs should be updated to describe that requirement.

Reply:
ok, that makes sense, so a `_tag` could be some other implementation, and that is valid for another use case. In that case, I also think `pass` is correct